unyt/tests/test_unit_registry.py

```diff
@@ -57,6 +57,29 @@ def test_modify_symbol_from_default_unit_registry():
         default_unit_registry.remove("cm")
 
 
+def test_modify_cache_clear():
+    # see https://github.com/yt-project/unyt/issues/473
+
+    ureg = UnitRegistry()
+    ureg.add("celery", 1.0, length)
+    u0 = Unit("m", registry=ureg)
+    Unit("celery", registry=ureg)
+
+    ureg.modify("celery", 0.5)
+    u2 = Unit("celery", registry=ureg)
+    assert 1.0 * u2 == 0.5 * u0
+
+    # TODO: test behaviour in u1 pre/post modification
```

Comment:
this line refers to the Unit created on after u0 (I guess linters removed the variable binding to `u1` because I didn't use it). The behaviour I'm getting with the branch at the moment is that the Unit object created before `ureg.modify` still uses the original base value even after modification of the registry. This seems to be in line with @AdamCohan's expectations, and seems consistent with the idea that `Unit` objects should be immutable, but I don't think that this case is currently *tested*. @jzuhone, what should be the expected behaviour in your opinion ?

Reply:
Let's try that again: @jzuhone

Reply:
The behavior you are seeing is correct.

Reply:
thank you ! I've updated the test to reflect this and rebased the branch, this is now ready for review !